### PR TITLE
add ThemeProvider component

### LINKED ISSUE #26: ThemeProvider
Thoughts on adding a ThemeProvider component that functions like the React version? I can try working on a PR this weekend if you'll accept this feature.

### PR DESCRIPTION
**Pull request by Jmeyering**
Use the inject/provide api to create a ThemeProvider component, injecting `theme` into the props of a StyledComponent.

I was unable to get the test that I wrote to pass, direction on what I'm doing wrong would be appreciated.

Would resolve #26.

### CODE REVIEW

**Comment by Jmeyering** [comment]
Any word on this? Anything I can do to move it along

File: src/test/props.test.js [hunk 1]
```diff
@@ -31,4 +32,32 @@ describe('props', () => {
...
+      render: function(createElement) {
+        return createElement(
+          ThemeProvider,
+          {
+            props: {
+              theme,
+            },
+          },
+          [
+            Comp
```
[the remark this answers is not in the document]
Comment (Jmeyering) [reply src/test/props.test.js]: Perfect! Thanks so much.

**Comment by Jmeyering** [comment]
Test updates Have been pushed up. Any word on getting this merged @liqueflies?

**Comment by JimmyLv** [comment]
Any update please?

**Pull request merged**